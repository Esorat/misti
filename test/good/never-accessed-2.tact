contract FieldTest {
    f1: Address;
    f2: Int as uint256 = 0;
    foreachmap: map<Address, Address>;

    init() {
        self.f1 = sender();
    }

    get fun use_f1(): Address {
        return self.f1;
    }

    fun forEachUse() {
        // No warning: foreachmap is used
        let foreachmap: map<Address, Address> = self.foreachmap;
        foreach (_, _ in foreachmap) {}
    }
}
