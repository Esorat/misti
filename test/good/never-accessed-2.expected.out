test/good/never-accessed-2.tact:3:5:
  2 |     f1: Address;
> 3 |     f2: Int as uint256 = 0;
          ^
  4 | 
Field is never used
Help: Consider creating a constant instead of field
See: https://nowarp.github.io/docs/misti/docs/detectors/NeverAccessedVariables